rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -221,7 +230,7 @@ where
         &self,
         logs: Vec<(Indexed<T>, LogMeta)>,
         stored_logs_metric: &GenericCounter<AtomicU64>,
-    ) -> Vec<(Indexed<T>, LogMeta)> {
+    ) -> Result<Vec<(Indexed<T>, LogMeta)>, Vec<(Indexed<T>, LogMeta)>> {
```

Comment:
sounds like what you really want to return here is a tuple of `(Vec<(Indexed<T>, LogMeta)>, bool)` where the bool indicates the success of the `store_logs` call. I think that'd be cleaner than this overloaded `Result` type

Reply:
Fixed